### PR TITLE
[Modules] Updated Network/VPNSites to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Network/VPNSites to new dependency approach

| Pipeline |
| - |
| [Network: VPN Sites](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.vpnsites.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/vpnSites/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,100 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/vpnSites/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/vpnSites/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,50 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/vpnSites/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/vpnSites/.test/common/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  name: managedIdentityName
+  location: location
+}
+
+resource virtualWan 'Microsoft.Network/virtualWans@2021-05-01' = {
+  name: virtualWANName
+  location: location
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (eriqua) [inline modules/Microsoft.Network/vpnSites/.test/common/dependencies.bicep:20]:
```suggestion
@description('The principal ID of the created managed identity.')
```

File: modules/Microsoft.Network/vpnSites/.test/common/dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+output managedIdentityPrincipalId string = managedIdentity.properties.principalId
+
+@description('The resource ID of the created Virtual WAN')
```
Comment (eriqua) [inline modules/Microsoft.Network/vpnSites/.test/common/dependencies.bicep:23]:
```suggestion
@description('The resource ID of the created Virtual WAN.')
```

File: modules/Microsoft.Network/vpnSites/.test/common/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(80)
+param resourceGroupName string = 'ms.network.vpnSites-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nvsicom'
```
Comment (eriqua) [inline modules/Microsoft.Network/vpnSites/.test/common/deploy.test.bicep:14]: Shouldn't this be
```suggestion
param serviceShort string = 'nvscom'
```

File: modules/Microsoft.Network/vpnSites/.test/min/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
```
Comment (eriqua) [inline modules/Microsoft.Network/vpnSites/.test/min/deploy.test.bicep:29]: So far I saw
```suggestion
  name: '${uniqueString(deployment().name, location)}-paramNested'
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**